rapids-cmake/export/export.cmake

```diff
@@ -187,6 +225,34 @@ function(rapids_export type project_name)
     set(_RAPIDS_PROJECT_NAMESPACE ${_RAPIDS_NAMESPACE})
   endif()
 
+  if(_RAPIDS_COMPONENTS AND NOT _RAPIDS_COMPONENTS_EXPORT_SET)
+    message(FATAL_ERROR "rapids_export(${type} ${project_name} is missing COMPONENTS_EXPORT_SET as COMPONENTS was provided."
+    )
+  endif()
+  if(_RAPIDS_COMPONENTS_EXPORT_SET AND NOT _RAPIDS_COMPONENTS)
+    message(FATAL_ERROR "rapids_export(${type} ${project_name} is missing COMPONENTS as COMPONENTS_EXPORT_SET was provided."
+    )
+  endif()
+
+  if(_RAPIDS_COMPONENTS AND _RAPIDS_COMPONENTS_EXPORT_SET)
+    include("${rapids-cmake-dir}/export/detail/component.cmake")
+    set(_RAPIDS_HAS_COMPONENTS TRUE)
+    set(_RAPIDS_COMPONENTS_EXPORT_SET_NICE_NAMES)
+
+    foreach(comp comp_export_set IN ZIP_LISTS _RAPIDS_COMPONENTS _RAPIDS_COMPONENTS_EXPORT_SET)
+      string(REGEX REPLACE "(${project_name}[-_])|([-_]?${comp}[-_]?)|([-_]?export[s]?)" ""
+                           nice_export_name "${comp_export_set}")
+      if(nice_export_name STREQUAL "")
+        string(PREPEND nice_export_name "${comp}")
+      else()
+        string(PREPEND nice_export_name "${comp}-")
+      endif()
+      rapids_export_component(${type} ${project_name} ${comp} ${comp_export_set}
+                              ${nice_export_name} ${_RAPIDS_PROJECT_NAMESPACE})
+      list(APPEND _RAPIDS_COMPONENTS_EXPORT_SET_NICE_NAMES ${nice_export_name})
```

Comment:
What is the purpose of tracking the nice names? If it is meant to be user-facing, we should name it without the underscore prefix and add it to the docs. If it is private, it seems like it doesn't actually serve any purpose.

Reply:
The nice name is used when generating the files names that the component information will be written too.

The nice names don't need to be known by users, but are generated so that the installed files all have 'reasonable' names

Reply:
Why do we need this list, then? Can't we simply remove it altogether?

Reply:
Yes it can be removed. Just realized it was left-over from a previous implementation of the `config.cmake`